cluster/cluster.go

```diff
@@ -10,10 +10,12 @@ import (
 
 type Cluster struct {
 	Registry *ServiceRegistry
+	etcd     *embed.Etcd
 }
 
 func Join(ctx context.Context, cfg Config) (*Cluster, error) {
-	if _, err := startEmbeddedEtcd(cfg.etcdConfig); err != nil {
+	e, err := startEmbeddedEtcd(cfg.etcdConfig)
```

Comment:
makes sense to me. If we ever have a node that faults and rejoins, calling `Join()` again will fail if the embedded etcd is still running.

Do you think it is worth it to skip starting the embedded etcd if one is running already, and silently continue?

Reply:
> If we ever have a node that faults and rejoins, calling Join() again will fail if the embedded etcd is still running.

This was added more for the purpose of having a graceful shutdown. I think in the case where the node dies and rejoins the embedded etcd would've died too since it's in a goroutine 

> Do you think it is worth it to skip starting the embedded etcd if one is running already, and silently continue?

That makes sense to me but I think we'd need to need the process that's using the port and confirm if it's etcd or not. This plus the not knowing if the existing etcd is healthy made it not seem worth doing